numpy/core/include/numpy/npy_math.h

```diff
@@ -184,29 +184,23 @@ NPY_INPLACE double npy_atan2(double x, double y);
 #define npy_fmod fmod
 #define npy_floor floor
 #define npy_expm1 expm1
+#define npy_log1p log1p
 #define npy_acosh acosh
+#define npy_asinh asinh
 #define npy_atanh atanh
 #define npy_rint rint
 #define npy_trunc trunc
 #define npy_exp2 exp2
 #define npy_frexp frexp
 #define npy_ldexp ldexp
+#define npy_copysign copysign
 #define npy_exp exp
 #define npy_sqrt sqrt
 #define npy_pow pow
 #define npy_modf modf
 
-#if defined(__arm64__) && defined(__APPLE__)
-/* due to a build problem with scipy, export these as functions */
-NPY_INPLACE double npy_asinh(double x);
-NPY_INPLACE double npy_copysign(double y, double x);
-NPY_INPLACE double npy_log1p(double x);
-#else
-#define npy_asinh asinh
-#define npy_copysign copysign
-#define npy_log1p log1p
-#endif
 double npy_nextafter(double x, double y);
```

Comment:
I restored this line which should not be removed

Reply:
Ah, thanks.  So that is why I had to add nextafter, but it was previously no issue.  I guess it could be excluded from the other file in that case, but seems it doesn't matter?

Reply:
It needs to be in both files for this branch. One is compiled into `libnpy_math.a`, this is the one that is removed from `main` but on this branch it is alive and well. The other implementation is exported from the main numpy namespace because of `<mumble>macos arm64 build linking problem</mumble>`

Reply:
@seberg @mattip 

This seems to be causing problems with the macosx m1 wheels, the linker complains of two symbols `_npy_nextafter`, which I suspect is because of long double being double on the m1 platform. This doesn't seem to be a problem in main.

Reply:
Arrgg, I guess that because `npy_nextafter` is already here, it should not have been included in `arm64_exports.c` for the backport...